aspnetcore/mvc/views/razor.md

```diff
@@ -442,41 +442,125 @@ public class _Views_Something_cshtml : RazorPage<dynamic>
 
 Later in this article, the section [Inspect the Razor C# class generated for a view](#inspect-the-razor-c-class-generated-for-a-view) explains how to view this generated class.
 
-<a name="using"></a>
+### @attribute
 
-### @using
+The `@attribute` directive adds the given attribute to the class of the generated page or view. The following example adds the `[Authorize]` attribute:
 
-The `@using` directive adds the C# `using` directive to the generated view:
+```cshtml
+@attribute [Authorize]
+```
 
-[!code-cshtml[](razor/sample/Views/Home/Contact9.cshtml)]
+::: moniker range=">= aspnetcore-3.0"
 
-### @model
+### @attributes
 
-The `@model` directive specifies the type of the model passed to a view:
+Elements can capture and render additional attributes in addition to the element's declared parameters using the `@attributes` Razor directive. This scenario is useful when defining an element that supports a variety of customizations.
+
+In the following example, additional attributes are captured in a dictionary and then applied onto the `<input>` element when the element is rendered:
 
 ```cshtml
-@model TypeNameOfModel
+<input @attributes="InputAttributes" />
+
+@code {
+    private Dictionary<string, object> InputAttributes { get; set; } =
+        new Dictionary<string, object>()
+        {
+            { "maxlength", "10" }, 
+            { "placeholder", "Placeholder text" }, 
+            { "required", "true" }, 
+            { "size", "50" }
+        };
 ```
 
-In an ASP.NET Core MVC app created with individual user accounts, the *Views/Account/Login.cshtml* view contains the following model declaration:
+The type of the parameter must be assignable from `Dictionary<string, object>` with string keys. Using `IEnumerable<KeyValuePair<string, object>>` and `IReadOnlyDictionary<string, object>` are also options in this scenario.
+
+The preceding code renders the following HTML markup:
+
+```html
+<input maxlength="10" placeholder="Placeholder text" required="required" size="50">
+```
+
+### @code
+
+The `@code` directive enables a Razor Page to add a C# code block to a view:
 
 ```cshtml
-@model LoginViewModel
+@functions { // C# Code }
 ```
 
-The class generated inherits from `RazorPage<dynamic>`:
+For example:
 
-```csharp
-public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel>
+```cshtml
```

Comment:
@NTaylorMullen @ajaybhargavb - did we end up supporting `@code` in all razor files? or just `.razor`?

Reply:
Only supported in .razor files

Reply:
Thanks @NTaylorMullen ... I'll get a commit in now for that.